pygaggle/data/segmentation.py

```diff
@@ -0,0 +1,53 @@
+import spacy
+import numpy as np
+from pygaggle.rerank.base import Text
+from typing import List
+
+
+class SegmentProcessor:
+
+    def __init__(self, documents: List[Text]):
+        """
+        'self.doc_end_indexes' stores the index of the last segment of each
+            document when 'segment()' converting a 'List[Text]' of documents into
+            'List[Text]' of segments. It will be used to split and group segments'
+            scores and feed the aggregated scores back to documents in 'aggregate()'
+        """
+        self.doc_end_indexes = [0]
+        self.documents = documents
+        self.aggregate_methods = {
+            "max": self._max_aggregate,
+            "mean": self._mean_aggregate
+        }
+
+    def segment(self, seg_size: int, stride: int) -> List[Text]:
+        segmented_doc, end_idx = [], 0
+        nlp = spacy.blank("en")
+        nlp.add_pipe(nlp.create_pipe("sentencizer"))
+        for document in self.documents:
+            doc = nlp(document.text[:1000000])
+            sentences = [sent.string.strip() for sent in doc.sents]
+            for i in range(0, len(sentences), stride):
+                segment_text = ' '.join(sentences[i:i + seg_size])
+                segmented_doc.append(Text(segment_text))
```

Comment:
I didn't maintain the doc id in segment level since it is not needed for the current functionality. I'll add it back, it can be good for debug or further usage.